### PR TITLE
Initial commit for AHA UX.

### PR DESCRIPTION
**Pull request by blackarmchair**
Views, Analytics, API, shaker logic; baseline app

### CODE REVIEW

File: app/shared/components/cmpt_popup.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+app.component('popup', {
```
Comment (LongBu) [inline app/shared/components/cmpt_popup.js]: @blackarmchair 
Is this component/*tmpl.html being used anywhere in the code?  I'm not sure if this has been superseded by the SlideCtrl.js?

File: app/shared/controllers/ResultsCtrl.js [hunk 2]
```diff
@@ -1,14 +1,31 @@
...
 '$stateParams',
-'ResultsCtrl',
+'DataService',
+'FormPrototype',
+'UserResult',
 function(
   $scope,
   $state,
   $stateParams,
-  DataService
```
Comment (LongBu) [inline app/shared/controllers/ResultsCtrl.js]: @blackarmchair 
Do we want to inject $analytics for any tracking on the results page? (eg: pageviews/actions/etc?)

File: app/shared/services/DataService.js [hunk 3]
```diff
@@ -7,4 +7,48 @@ function(
...
+  };
+
+  function getEnvironmentPath() {
+    var dev  = /dev/g.test(window.location.hostname);
+    var beta = /test/g.test(window.location.hostname);
+    var prod = /m\.bsh\.io|^nep\.nmi\.cc/g.test(window.location.hostname);
+    var path;
+
+    if ( dev )       path = "https://dev.bsh.io/nep/microsite/sandbox/";
+    else if ( beta ) path = "https://test.bsh.io/nep/microsite/";
```
Comment (LongBu) [inline app/shared/services/DataService.js]: @blackarmchair 
might want to consider the fileserver/webserver  dev spaces, too for future development

File: data/signup.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  {
+    "display":     "",
+    "placeholder": "Email",
+    "name":        "email",
+    "type":        "email",
+    "validator":   "email",
+    "validMsg":    "Please enter a valid email address.",
+    "minLength":   "6",
+    "maxLength":   "20",
```
Comment (LongBu) [inline data/signup.json]: @blackarmchair 
Might want to consider a longer maxLength for the email/name fields.  Some South Indian names can get real long (outliers, but possible to come up)

File: assets/css/partials/transitions.scss [hunk 5]
```diff
@@ -197,95 +197,95 @@
...
-  //     }
-  //   }
-  //   .big-entrance {
-  //     @include animation('big-entrance 2s ease-out');
-  //     &.no-delay {
-  //       @include animation('big-entrance 1s ease-out');
-  //     }
-  //     visibility: visible !important;
-  //   }
+  // Big entrance
```
Comment (LongBu) [inline assets/css/partials/transitions.scss]: @blackarmchair 
Remove commented out stuff unless needed.  Granted this is removed through sass/minification/etc., but makes it cleaner to read.

**Comment by LongBu** [comment]
@blackarmchair 
Should likely not include the minified/uglified/packaged files in the branch (eg: styles.min.css(.map)/app.xyz.js/etc.

File: assets/css/views/thanks.scss [hunk 6]
```diff
@@ -1,2 +1,29 @@
 
-#thanksTmpl {}
+#thanksTmpl {
```
Comment (LongBu) [inline assets/css/views/thanks.scss]: @blackarmchair 
This might not be possible, but wondering if *.scss files can inherit/import from other files.  Noticing a lot of the same lines between thanks.scss and results.scss.  This isn't a must, but it could help out maybe in the long term to not repeat lines.

File: app/views/cmpt_input.tmpl.html [hunk 7]
```diff
@@ -1,10 +1,13 @@
 
 <div ng-switch="$ctrl.type" id="formWrapper">
 
+  <!-- Text Label -->
+  <div class="form-group" ng-switch-when="label">
+    <p><strong>{{ $ctrl.display }}</strong> {{ $ctrl.placeholder }}</p>
```
Comment (LongBu) [inline app/views/cmpt_input.tmpl.html]: @blackarmchair 
Removing inline styles likely would be ideal

File: data/slides.json [hunk 8]
```diff
@@ -0,0 +1,113 @@
+{
+  "top": [
+    {
+      "text": "Where are you in your career?",
+      "subtext": "(Swipe to Select)",
+      "style": "bg-image",
+      "image": "1.png",
```
Comment (LongBu) [inline data/slides.json]: @blackarmchair 
Are the images in here relative to the imagePath?

File: app/shared/components/cmpt_popup.js (hunk 1 above)
Comment (blackarmchair) [inline app/shared/components/cmpt_popup.js]: @LongBu 

Yes, it's being used via [transclusion](https://en.wikipedia.org/wiki/Transclusion). That is, the template itself is a meta-template that designates areas where more specific content can be entered.

For example <popup-content> has a default value along the lines of "oops, something went wrong" because the expectation is that anywhere the tag is used we'll fill it in with content specific to whatever template is referencing it.

File: app/shared/controllers/ResultsCtrl.js (hunk 2 above)
Comment (blackarmchair) [inline app/shared/controllers/ResultsCtrl.js]: We don't need to inject $analytics into the controller unless the controller itself is going to use it. If a template associated with the controller (e.g. results.tmpl.html) attempts to use a directive that references the analytics plugin (e.g. analytics-on) we're covered because that directive already has the $analytics provider injected into it.

**tl;dr We only need to inject it if we're going to write `$analytics` somewhere in the controller.**

File: app/shared/components/cmpt_popup.js (hunk 1 above)
Comment (LongBu) [inline app/shared/components/cmpt_popup.js]: @blackarmchair 
Cool, thanks for clearing this up.

File: app/shared/services/DataService.js (hunk 3 above)
Comment (blackarmchair) [inline app/shared/services/DataService.js]: @LongBu 

This is a good idea; what's your suggestion for improving this?

File: data/signup.json (hunk 4 above)
Comment (blackarmchair) [inline data/signup.json]: @LongBu: sure, you thinking 255 to match the SQL column?

File: app/shared/controllers/ResultsCtrl.js (hunk 2 above)
Comment (LongBu) [inline app/shared/controllers/ResultsCtrl.js]: @blackarmchair 
Makes sense

File: assets/css/partials/transitions.scss (hunk 5 above)
Comment (blackarmchair) [inline assets/css/partials/transitions.scss]: @LongBu 

I leave it in because I want to have the option of "turning-on" the animations should I need them. They're very expensive to leave in the stylesheet if they're not being used because SASS vendor-prefixes each one.

File: app/shared/services/DataService.js (hunk 3 above)
Comment (LongBu) [inline app/shared/services/DataService.js]: @blackarmchair 
Regular expression/str comparison to check if sandbox is in the path =>fileserverdev, else webserverdev?

File: assets/css/views/thanks.scss (hunk 6 above)
Comment (blackarmchair) [inline assets/css/views/thanks.scss]: @LongBu 

It definitely can; I could likely make these files more DRY.

File: data/signup.json (hunk 4 above)
Comment (LongBu) [inline data/signup.json]: @blackarmchair 
255 sounds good to me.

File: assets/css/partials/transitions.scss (hunk 5 above)
Comment (LongBu) [inline assets/css/partials/transitions.scss]: @blackarmchair 
That's fine I guess given that it's dropped via grunt/Sass.

File: app/views/cmpt_input.tmpl.html (hunk 7 above)
Comment (blackarmchair) [inline app/views/cmpt_input.tmpl.html]: @LongBu 

I take your point but this isn't really an inline style.

Do you think `<span class="bold">...</span>` would be better? It's the same amount of markup and the same amount of style information; I just think `<strong>...</strong>` makes more sense semantically.

Thoughts?

File: assets/css/views/thanks.scss (hunk 6 above)
Comment (LongBu) [inline assets/css/views/thanks.scss]: @blackarmchair 
Dry is a good aim, but if things divert make it separate.  Thanks!

File: app/shared/services/DataService.js (hunk 3 above)
Comment (blackarmchair) [inline app/shared/services/DataService.js]: @LongBu 

That leaves us unable to differentiate between beta and prod though? Is that acceptable?

File: assets/css/partials/transitions.scss (hunk 5 above)
Comment (blackarmchair) [inline assets/css/partials/transitions.scss]: @LongBu 

Yeah, that was my thinking as well. We could leave the full version in the boilerplate app and eliminate the unused animations from each project too for ease-of-use.

File: app/views/cmpt_input.tmpl.html (hunk 7 above)
Comment (LongBu) [inline app/views/cmpt_input.tmpl.html]: @blackarmchair 
If it's only one line I guess I'm being a little too OCD, but if it get's to be substantial I'd say reconsider this.  Ignore this.

File: app/shared/services/DataService.js (hunk 3 above)
Comment (LongBu) [inline app/shared/services/DataService.js]: @blackarmchair 
Maybe use window.location.href instead of window.location.hostname to parse out the domain/subaddress stuff?  Thoughts?
